dds/DCPS/MultiTopicDataReaderBase.cpp

```diff
@@ -173,10 +172,11 @@ void MultiTopicDataReaderBase::data_available(DDS::DataReader_ptr reader)
   if (rc == RETCODE_NO_DATA) {
     return;
   } else if (rc != RETCODE_OK) {
-    ostringstream rc_ss;
-    rc_ss << rc;
+    OPENDDS_STRING rc_ss;
+    rc_ss.reserve(sizeof(ReturnCode_t));
+    rc_ss += rc;
```

Comment:
sizeof() is number of bytes in memory, not number of characters in the string
string += int is appending the character value of the integer